tests/genn_tests/test_iaf_psc_exp.py

```diff
@@ -0,0 +1,182 @@
+        current_input.target_var = "I_stim"
+        model.build()
+        model.load(num_recording_timesteps=SIM_TIMESTEPS)
+
+        V_m_log = []
+        while model.timestep < SIM_TIMESTEPS:
+            model.step_time()
+            neuron_pop.vars["V_m"].pull_from_device()
+            V_m_log.append(neuron_pop.vars["V_m"].values)
+
+        model.pull_recording_buffers_from_device()
+
+        spike_times, spike_ids = neuron_pop.spike_recording_data[0]
+
+        if TEST_PLOTS:
+            fig, ax = plt.subplots()
+
+            ax.scatter(spike_times, spike_ids, s=1)
+            ax.set_ylabel("Neuron ID")
+            ax.set_xlim((0, SIM_TIMESTEPS * TIMESTEP))
+
+            fig.savefig("/tmp/genn_iaf_psc_exp_current_injection.png")
+            plt.close(fig)
+
+        assert len(spike_times) > 10
+
+    def test_genn_iaf_psc_exp_postsynaptic_response(self):
+        from nestmlmodule.iaf_psc_exp_neuron_nestml import iaf_psc_exp_neuron_nestml_model
+
+        # Simulation timestep of model in ms
+        TIMESTEP = 1.0
+        SIM_TIMESTEPS = 100
+
+        iaf_psc_exp_init = {"V_m": -70.0,
+                            "refr_t": 0.,
+                            "I_syn_exc": 0.,
+                            "I_syn_inh": 0.}
+
+        iaf_psc_exp_params = {"C_m": 250.,
+                              "tau_m": 10.,
+                              "tau_syn_inh": 2.,
+                              "tau_syn_exc": 2.,
+                              "refr_T": 2.,
+                              "E_L": -70.,
+                              "V_reset": -70.,
+                              "V_th": 55.,
+                              "I_e": 0.}
+
+        model = GeNNModel("float", "test_genn_iaf_psc_exp_postsynaptic_response")
+        model.dt = TIMESTEP
+        neuron_pop = model.add_neuron_population("neuron0", 1,
+                                                 iaf_psc_exp_neuron_nestml_model, iaf_psc_exp_params, iaf_psc_exp_init)
+        neuron_pop.spike_recording_enabled = True
+
+        poisson_gen_pop = model.add_neuron_population("my_poisson_gen", 1,
+                                                      "Poisson", {"rate": 200.}, {"timeStepToSpike": 1})
+        poisson_gen_pop.spike_recording_enabled = True
+        syn_pop = model.add_synapse_population("synapse_0_1", "DENSE",
+                                               poisson_gen_pop, neuron_pop,
+                                               init_weight_update("StaticPulse", {}, {"g": 10000000.}),
+                                               init_postsynaptic("DeltaCurr"))
+        syn_pop.post_target_var = "exc_spikes"
+
+        model.build()
+        model.load(num_recording_timesteps=SIM_TIMESTEPS)
+
+        V_m_log = []
+        while model.timestep < SIM_TIMESTEPS:
+            model.step_time()
+            neuron_pop.vars["V_m"].pull_from_device()
+            V_m_log.append(neuron_pop.vars["V_m"].values)
+
+        model.pull_recording_buffers_from_device()
```

Comment:
Since GeNN generates code for NVIDIA GPUs, where does the simulation run in this case?

Reply:
It runs on GeNN's CPU backend instead.